crates/sui-rosetta/src/construction.rs

```diff
@@ -239,6 +238,12 @@ pub async fn metadata(
     let option = request.options.ok_or(Error::MissingMetadata)?;
     let budget = option.budget;
     let sender = option.internal_operation.sender();
+    let currency = match &option.internal_operation {
+        InternalOperation::PayCoin { currency, .. } => Some(currency.clone()),
+        _ => None,
+    };
+    let coin_type = currency.clone().map(|c| c.metadata.coin_type);
```

Comment:
Minor: Instead of cloning the whole `currency` struct, you can use `currency.as_ref().map(|c| c.metadata.coin_type.clone())`.